--- gdb-7.0/gdb/symfile.c~	2009-12-08 08:19:05.000000000 +0200
+++ gdb-7.0/gdb/symfile.c	2009-12-08 08:47:21.738081248 +0200
@@ -2236,8 +2236,8 @@
 	fprintf_unfiltered (gdb_stdlog,
 			    _("Missing separate debuginfo for %s\n"), binary);
         if (debug != NULL)
-	  fprintf_unfiltered (gdb_stdlog, _("Try: %s %s\n"),
-			      "yum --enablerepo='*-debuginfo' install", debug);
+	  fprintf_unfiltered (gdb_stdlog, _("Try to install package that provides `%s' file\n"),
+			      debug);
       }
 }
 
