--- gdb-7.5.0.20120926/gdb/elfread.c.orig	2012-11-18 20:50:42.844619396 +0100
+++ gdb-7.5.0.20120926/gdb/elfread.c	2012-11-18 20:52:37.756554669 +0100
@@ -2237,8 +2237,8 @@
 	fprintf_unfiltered (gdb_stdlog,
 			    _("Missing separate debuginfo for %s\n"), binary);
         if (debug != NULL)
-	  fprintf_unfiltered (gdb_stdlog, _("Try: %s %s\n"),
-			      "yum --enablerepo='*debug*' install", debug);
+	  fprintf_unfiltered (gdb_stdlog, _("Try to install package that provides `%s' file\n"),
+			      debug);
       }
 }
 
