--- gdb-7.7.1/gdb/valprint.c.orig	2014-06-04 16:13:46.353948386 +0200
+++ gdb-7.7.1/gdb/valprint.c	2014-06-04 16:15:55.599699077 +0200
@@ -108,12 +108,12 @@
 struct value_print_options user_print_options =
 {
   Val_prettyformat_default,	/* prettyformat */
-  0,				/* prettyformat_arrays */
-  0,				/* prettyformat_structs */
-  0,				/* vtblprint */
+  1,				/* prettyformat_arrays */
+  1,				/* prettyformat_structs */
+  1,				/* vtblprint */
   1,				/* unionprint */
   1,				/* addressprint */
-  0,				/* objectprint */
+  1,				/* objectprint */
   PRINT_MAX_DEFAULT,		/* print_max */
   10,				/* repeat_count_threshold */
   0,				/* output_format */
