diff -urN gdb-5.1.org/sim/erc32/configure gdb-5.1/sim/erc32/configure
--- gdb-5.1.org/sim/erc32/configure	Mon Jan  7 00:21:33 2002
+++ gdb-5.1/sim/erc32/configure	Mon Jan  7 00:23:01 2002
@@ -3572,7 +3572,7 @@
   echo $ac_n "(cached) $ac_c" 1>&6
 else
   ac_save_LIBS="$LIBS"
-LIBS="-ltermcap  $LIBS"
+LIBS="-lncurses $LIBS"
 cat > conftest.$ac_ext <<EOF
 #line 3578 "configure"
 #include "confdefs.h"
@@ -3596,7 +3596,7 @@
 fi
 if eval "test \"`echo '$ac_cv_lib_'$ac_lib_var`\" = yes"; then
   echo "$ac_t""yes" 1>&6
-  TERMCAP=-ltermcap
+  TERMCAP=-lncurses
 else
   echo "$ac_t""no" 1>&6
 TERMCAP=""
diff -urN gdb-5.1.org/sim/erc32/configure.in gdb-5.1/sim/erc32/configure.in
--- gdb-5.1.org/sim/erc32/configure.in	Mon Jan  7 00:21:33 2002
+++ gdb-5.1/sim/erc32/configure.in	Mon Jan  7 00:22:16 2002
@@ -17,7 +17,7 @@
 if test x$sim_cv_os_cygwin = xyes; then
   TERMCAP='`if test -r ../../libtermcap/libtermcap.a; then echo ../../libtermcap/libtermcap.a; else echo -ltermcap; fi` -luser32'
 else
-  AC_CHECK_LIB(termcap, main, TERMCAP=-ltermcap, TERMCAP="")
+  AC_CHECK_LIB(ncurses, main, TERMCAP=-lncurses, TERMCAP="")
 fi
 AC_SUBST(TERMCAP)
 
