--- gdb-6.1.1/gdb/ada-lang.c.orig	2004-06-23 17:49:34.000000000 +0200
+++ gdb-6.1.1/gdb/ada-lang.c	2004-06-23 18:58:04.000000000 +0200
@@ -10141,7 +10141,7 @@
   &ada_exp_descriptor,
   parse,
   ada_error,
-  resolve,
+  /* resolve, CVS gdb */
   ada_printchar,		/* Print a character constant */
   ada_printstr,			/* Function to print string constant */
   emit_char,			/* Function to print single char (not used) */
