diff -urN gdb-6.0.org/gdb/gdbserver/Makefile.in gdb-6.0/gdb/gdbserver/Makefile.in
--- gdb-6.0.org/gdb/gdbserver/Makefile.in	2003-10-09 19:42:41.000000000 +0200
+++ gdb-6.0/gdb/gdbserver/Makefile.in	2003-10-09 19:48:08.000000000 +0200
@@ -76,7 +76,7 @@
 
 # Where is the source dir for the READLINE library?  Traditionally in .. or .
 # (For the binary library built from it, we use ${READLINE_DIR}${subdir}.)
-READLINE_DIR = ${srcdir}/../readline
+READLINE_DIR = /usr/lib
 READLINE_DEP = $$(READLINE_DIR)
 
 # All the includes used for CFLAGS and for lint.
diff -urN gdb-6.0.org/gdb/Makefile.in gdb-6.0/gdb/Makefile.in
--- gdb-6.0.org/gdb/Makefile.in	2003-10-09 19:42:39.000000000 +0200
+++ gdb-6.0/gdb/Makefile.in	2003-10-09 19:48:08.000000000 +0200
@@ -116,10 +116,10 @@
 BFD_CFLAGS = -I$(BFD_DIR) -I$(BFD_SRC)
 
 # Where is the READLINE library?  Typically in ../readline.
-READLINE_DIR = ../readline
-READLINE = $(READLINE_DIR)/libreadline.a
-READLINE_SRC = $(srcdir)/$(READLINE_DIR)
-READLINE_CFLAGS = -I$(READLINE_SRC)/..
+READLINE_DIR = /usr/include/readline
+READLINE = -lreadline
+READLINE_SRC = /usr/include/readline
+READLINE_CFLAGS = -I/usr/include/readline
 
 WARN_CFLAGS = @WARN_CFLAGS@
 WERROR_CFLAGS = @WERROR_CFLAGS@
@@ -369,7 +369,7 @@
 	$(XM_CLIBS) $(TM_CLIBS) $(NAT_CLIBS) $(GDBTKLIBS) @LIBS@ \
 	$(LIBICONV) \
 	$(LIBIBERTY) $(WIN32LIBS)
-CDEPS = $(XM_CDEPS) $(TM_CDEPS) $(NAT_CDEPS) $(SIM) $(BFD) $(READLINE) \
+CDEPS = $(XM_CDEPS) $(TM_CDEPS) $(NAT_CDEPS) $(SIM) $(BFD) \
 	$(OPCODES) $(INTL_DEPS) $(LIBIBERTY) $(CONFIG_DEPS)
 
 ADD_FILES = $(XM_ADD_FILES) $(TM_ADD_FILES) $(NAT_ADD_FILES)
--- gdb-6.3/Makefile.in.orig	2004-11-08 17:21:22.000000000 +0100
+++ gdb-6.3/Makefile.in	2004-11-19 23:30:12.276705120 +0100
@@ -692,7 +692,6 @@
     maybe-configure-perl \
     maybe-configure-prms \
     maybe-configure-rcs \
-    maybe-configure-readline \
     maybe-configure-release \
     maybe-configure-recode \
     maybe-configure-sed \
@@ -793,7 +792,6 @@
     maybe-all-perl \
     maybe-all-prms \
     maybe-all-rcs \
-    maybe-all-readline \
     maybe-all-release \
     maybe-all-recode \
     maybe-all-sed \
@@ -889,7 +887,6 @@
     maybe-info-perl \
     maybe-info-prms \
     maybe-info-rcs \
-    maybe-info-readline \
     maybe-info-release \
     maybe-info-recode \
     maybe-info-sed \
@@ -981,7 +978,6 @@
     maybe-dvi-perl \
     maybe-dvi-prms \
     maybe-dvi-rcs \
-    maybe-dvi-readline \
     maybe-dvi-release \
     maybe-dvi-recode \
     maybe-dvi-sed \
@@ -1073,7 +1069,6 @@
     maybe-TAGS-perl \
     maybe-TAGS-prms \
     maybe-TAGS-rcs \
-    maybe-TAGS-readline \
     maybe-TAGS-release \
     maybe-TAGS-recode \
     maybe-TAGS-sed \
@@ -1165,7 +1160,6 @@
     maybe-install-info-perl \
     maybe-install-info-prms \
     maybe-install-info-rcs \
-    maybe-install-info-readline \
     maybe-install-info-release \
     maybe-install-info-recode \
     maybe-install-info-sed \
@@ -1257,7 +1251,6 @@
     maybe-installcheck-perl \
     maybe-installcheck-prms \
     maybe-installcheck-rcs \
-    maybe-installcheck-readline \
     maybe-installcheck-release \
     maybe-installcheck-recode \
     maybe-installcheck-sed \
@@ -1349,7 +1342,6 @@
     maybe-mostlyclean-perl \
     maybe-mostlyclean-prms \
     maybe-mostlyclean-rcs \
-    maybe-mostlyclean-readline \
     maybe-mostlyclean-release \
     maybe-mostlyclean-recode \
     maybe-mostlyclean-sed \
@@ -1441,7 +1433,6 @@
     maybe-clean-perl \
     maybe-clean-prms \
     maybe-clean-rcs \
-    maybe-clean-readline \
     maybe-clean-release \
     maybe-clean-recode \
     maybe-clean-sed \
@@ -1533,7 +1524,6 @@
     maybe-distclean-perl \
     maybe-distclean-prms \
     maybe-distclean-rcs \
-    maybe-distclean-readline \
     maybe-distclean-release \
     maybe-distclean-recode \
     maybe-distclean-sed \
@@ -1625,7 +1615,6 @@
     maybe-maintainer-clean-perl \
     maybe-maintainer-clean-prms \
     maybe-maintainer-clean-rcs \
-    maybe-maintainer-clean-readline \
     maybe-maintainer-clean-release \
     maybe-maintainer-clean-recode \
     maybe-maintainer-clean-sed \
@@ -1774,7 +1763,6 @@
     maybe-check-perl \
     maybe-check-prms \
     maybe-check-rcs \
-    maybe-check-readline \
     maybe-check-release \
     maybe-check-recode \
     maybe-check-sed \
@@ -1885,7 +1873,6 @@
     maybe-install-perl \
     maybe-install-prms \
     maybe-install-rcs \
-    maybe-install-readline \
     maybe-install-release \
     maybe-install-recode \
     maybe-install-sed \
@@ -1953,7 +1940,6 @@
     maybe-install-perl \
     maybe-install-prms \
     maybe-install-rcs \
-    maybe-install-readline \
     maybe-install-release \
     maybe-install-recode \
     maybe-install-sed \
@@ -32695,7 +32681,6 @@
 all-gdb: maybe-all-libiberty
 all-gdb: maybe-all-opcodes
 all-gdb: maybe-all-bfd
-all-gdb: maybe-all-readline
 all-gdb: maybe-all-build-bison
 all-gdb: maybe-all-build-byacc
 all-gdb: maybe-all-sim
@@ -32928,7 +32913,6 @@
 all-sim: maybe-all-libiberty
 all-sim: maybe-all-bfd
 all-sim: maybe-all-opcodes
-all-sim: maybe-all-readline
 all-sim: maybe-configure-gdb
 all-fastjar: maybe-all-zlib
 all-fastjar: maybe-all-build-texinfo
