--- gdb-14.1/gdb/valprint.c.orig	2014-06-04 16:13:46.353948386 +0200
+++ gdb-14.1/gdb/valprint.c	2014-06-04 16:15:55.599699077 +0200
@@ -106,13 +106,13 @@ static void val_print_type_code_flags (s
 struct value_print_options user_print_options =
 {
   Val_prettyformat_default,	/* prettyformat */
-  false,			/* prettyformat_arrays */
-  false,			/* prettyformat_structs */
-  false,			/* vtblprint */
+  true,				/* prettyformat_arrays */
+  true,				/* prettyformat_structs */
+  true,				/* vtblprint */
   true,				/* unionprint */
   true,				/* addressprint */
   false,			/* nibblesprint */
-  false,			/* objectprint */
+  true,				/* objectprint */
   PRINT_MAX_DEFAULT,		/* print_max */
   PRINT_MAX_CHARS_DEFAULT,	/* print_max_chars */
   10,				/* repeat_count_threshold */
