--- gdb-6.1/gdb/lin-lwp.c.orig	2003-10-08 22:05:56.000000000 +0200
+++ gdb-6.1/gdb/lin-lwp.c	2004-04-25 01:39:57.838162954 +0200
@@ -22,13 +22,13 @@
 
 #include "gdb_assert.h"
 #include "gdb_string.h"
+#include <sys/ptrace.h>
 #include <errno.h>
 #include <signal.h>
 #ifdef HAVE_TKILL_SYSCALL
 #include <unistd.h>
 #include <sys/syscall.h>
 #endif
-#include <sys/ptrace.h>
 #include "gdb_wait.h"
 
 #include "gdbthread.h"
--- gdb-6.1/gdb/sparc-nat.c.orig	2004-04-24 23:48:09.000000000 +0000
+++ gdb-6.1/gdb/sparc-nat.c	2004-04-25 00:18:12.000000000 +0000
@@ -23,7 +23,7 @@
  * between <asm/reg.h> and <sys/ucontext.h> (included in <signal.h>)
  * gdb needs <asm/reg.h>, so this is workaround to avoid including
  * <sys/ucontext.h> contents */
-#define _SYS_UCONTEXT_H 1
+/*#define _SYS_UCONTEXT_H 1*/
 
 #include "defs.h"
 #include "inferior.h"
