--- gdb-5.3/Makefile.in.orig	Tue Dec 10 22:30:40 2002
+++ gdb-5.3/Makefile.in	Sat Jan 11 13:22:38 2003
@@ -566,7 +566,6 @@
 	all-perl \
 	all-prms \
 	all-rcs \
-	all-readline \
 	all-release \
 	all-recode \
 	all-sed \
@@ -637,7 +636,6 @@
 	check-perl \
 	check-prms \
 	check-rcs \
-	check-readline \
 	check-recode \
 	check-sed \
 	check-send-pr \
@@ -707,7 +705,6 @@
 	install-perl \
 	install-prms \
 	install-rcs \
-	install-readline \
 	install-recode \
 	install-sed \
 	install-send-pr \
@@ -868,7 +865,6 @@
 	clean-perl \
 	clean-prms \
 	clean-rcs \
-	clean-readline \
 	clean-release \
 	clean-recode \
 	clean-sed \
@@ -1677,7 +1673,7 @@
 all-gcc: all-bison all-byacc all-binutils all-gas all-ld all-zlib
 all-bootstrap: all-libiberty all-texinfo all-bison all-byacc all-binutils all-gas all-ld all-zlib
 GDB_TK = all-tk all-tcl all-itcl all-tix all-libgui
-all-gdb: all-libiberty all-opcodes all-bfd all-mmalloc all-readline all-bison all-byacc all-sim $(gdbnlmrequirements) $(GDB_TK)
+all-gdb: all-libiberty all-opcodes all-bfd all-mmalloc all-bison all-byacc all-sim $(gdbnlmrequirements) $(GDB_TK)
 all-gettext:
 all-gnuserv:
 configure-target-gperf: $(ALL_GCC_CXX)
@@ -1724,13 +1720,12 @@
 configure-target-qthreads: $(ALL_GCC_C)
 all-target-qthreads: configure-target-qthreads
 all-rcs:
-all-readline:
 all-recode: all-libiberty
 all-sed: all-libiberty
 all-send-pr: all-prms
 all-shellutils:
 all-sid: all-tcl all-tk
-all-sim: all-libiberty all-bfd all-opcodes all-readline
+all-sim: all-libiberty all-bfd all-opcodes
 all-snavigator: all-tcl all-tk all-itcl all-tix all-db all-grep all-libgui
 all-tar: all-libiberty
 all-tcl:
@@ -2005,7 +2000,7 @@
 		SUPPORT_FILES="$(GNATS_SUPPORT_DIRS)"
 
 .PHONY: gdb.tar.bz2
-GDB_SUPPORT_DIRS= bfd include libiberty mmalloc opcodes readline sim utils intl
+GDB_SUPPORT_DIRS= bfd include libiberty mmalloc opcodes sim utils intl
 gdb.tar.bz2: $(DIST_SUPPORT) $(GDB_SUPPORT_DIRS) gdb
 	$(MAKE) -f Makefile.in gdb-taz TOOL=gdb \
 		MD5PROG="$(MD5PROG)" \
diff -ur gdb-5.1.orig/gdb/Makefile.in gdb-5.1/gdb/Makefile.in
--- gdb-5.1.orig/gdb/Makefile.in	Sun Nov 18 06:08:36 2001
+++ gdb-5.1/gdb/Makefile.in	Tue Jan 22 23:54:56 2002
@@ -120,10 +120,10 @@
 BFD_CFLAGS = -I$(BFD_DIR) -I$(BFD_SRC)
 
 # Where is the READLINE library?  Typically in ../readline.
-READLINE_DIR = ../readline
-READLINE = $(READLINE_DIR)/libreadline.a
-READLINE_SRC = $(srcdir)/$(READLINE_DIR)
-READLINE_CFLAGS = -I$(READLINE_SRC)/..
+READLINE_DIR = /usr/include/readline
+READLINE = -lreadline
+READLINE_SRC = /usr/include/readline
+READLINE_CFLAGS = -I/usr/include/readline
 
 WARN_CFLAGS = @WARN_CFLAGS@
 WERROR_CFLAGS = @WERROR_CFLAGS@
diff -ur gdb-5.1.orig/gdb/defs.h gdb-5.1/gdb/defs.h
--- gdb-5.1.orig/gdb/defs.h	Thu Aug  2 23:36:51 2001
+++ gdb-5.1/gdb/defs.h	Tue Jan 22 23:59:50 2002
@@ -800,7 +800,7 @@
 
 /* From readline (but not in any readline .h files).  */
 
-extern char *tilde_expand (char *);
+extern char *tilde_expand (const char *);
 
 /* Control types for commands */
 
diff -ur gdb-5.1.orig/gdb/event-top.c gdb-5.1/gdb/event-top.c
--- gdb-5.1.orig/gdb/event-top.c	Sat Jul 14 20:59:07 2001
+++ gdb-5.1/gdb/event-top.c	Tue Jan 22 23:59:50 2002
@@ -32,6 +32,7 @@
 #include "gdbcmd.h"
 
 /* readline include files */
+#define savestring
 #include <readline/readline.h>
 #include <readline/history.h>
 
diff -ur gdb-5.1.orig/gdb/gdbserver/Makefile.in gdb-5.1/gdb/gdbserver/Makefile.in
--- gdb-5.1.orig/gdb/gdbserver/Makefile.in	Tue Mar  6 09:21:43 2001
+++ gdb-5.1/gdb/gdbserver/Makefile.in	Tue Jan 22 23:59:50 2002
@@ -88,7 +88,7 @@
 
 # Where is the source dir for the READLINE library?  Traditionally in .. or .
 # (For the binary library built from it, we use ${READLINE_DIR}${subdir}.)
-READLINE_DIR = ${srcdir}/../readline
+READLINE_DIR = /usr/lib
 READLINE_DEP = $$(READLINE_DIR)
 
 # All the includes used for CFLAGS and for lint.
diff -ur gdb-5.1.orig/gdb/top.c gdb-5.1/gdb/top.c
--- gdb-5.1.orig/gdb/top.c	Sun Oct 21 21:41:19 2001
+++ gdb-5.1/gdb/top.c	Tue Jan 22 23:59:50 2002
@@ -42,6 +42,7 @@
 #include "serial.h"
 
 /* readline include files */
+#define savestring
 #include <readline/readline.h>
 #include <readline/history.h>
 
diff -ur gdb-5.1.orig/gdb/tracepoint.c gdb-5.1/gdb/tracepoint.c
--- gdb-5.1.orig/gdb/tracepoint.c	Thu Aug  2 14:05:57 2001
+++ gdb-5.1/gdb/tracepoint.c	Tue Jan 22 23:59:50 2002
@@ -40,6 +40,7 @@
 #include "ax-gdb.h"
 
 /* readline include files */
+#define savestring
 #include <readline/readline.h>
 #include <readline/history.h>
 
diff -ur gdb-5.1.orig/gdb/utils.c gdb-5.1/gdb/utils.c
--- gdb-5.1.orig/gdb/utils.c	Fri Nov  2 23:06:29 2001
+++ gdb-5.1/gdb/utils.c	Tue Jan 22 23:59:50 2002
@@ -54,6 +54,7 @@
 
 #include "inferior.h" /* for signed_pointer_to_address */
 
+#define savestring
 #include <readline/readline.h>
 
 #ifndef MALLOC_INCOMPATIBLE
--- gdb-5.3/gdb/tui/tuiWin.c.orig	Fri Oct 25 21:46:41 2002
+++ gdb-5.3/gdb/tui/tuiWin.c	Sat Jan 11 13:24:57 2003
@@ -633,10 +633,10 @@
 tuiResizeAll (void)
 {
   int heightDiff, widthDiff;
-  extern int screenheight, screenwidth;		/* in readline */
+  extern int _rl_screenheight, _rl_screenwidth;		/* in readline */
 
-  widthDiff = screenwidth - termWidth ();
-  heightDiff = screenheight - termHeight ();
+  widthDiff = _rl_screenwidth - termWidth ();
+  heightDiff = _rl_screenheight - termHeight ();
   if (heightDiff || widthDiff)
     {
       TuiLayoutType curLayout = currentLayout ();
@@ -650,8 +650,8 @@
       if (winWithFocus != cmdWin)
 	keypad (cmdWin->generic.handle, FALSE);
       tui_update_gdb_sizes ();
-      setTermHeightTo (screenheight);
-      setTermWidthTo (screenwidth);
+      setTermHeightTo (_rl_screenheight);
+      setTermWidthTo (_rl_screenwidth);
       if (curLayout == SRC_DISASSEM_COMMAND ||
 	curLayout == SRC_DATA_COMMAND || curLayout == DISASSEM_DATA_COMMAND)
 	numWinsDisplayed++;
@@ -678,8 +678,8 @@
 	  if (heightDiff == 0)
 	    newHeight = firstWin->generic.height;
 	  else if ((firstWin->generic.height + splitDiff) >=
-		   (screenheight - MIN_CMD_WIN_HEIGHT - 1))
-	    newHeight = screenheight - MIN_CMD_WIN_HEIGHT - 1;
+		   (_rl_screenheight - MIN_CMD_WIN_HEIGHT - 1))
+	    newHeight = _rl_screenheight - MIN_CMD_WIN_HEIGHT - 1;
 	  else if ((firstWin->generic.height + splitDiff) <= 0)
 	    newHeight = MIN_WIN_HEIGHT;
 	  else
@@ -688,7 +688,7 @@
 	  _makeInvisibleAndSetNewHeight (firstWin, newHeight);
 	  cmdWin->generic.origin.y = locator->origin.y + 1;
 	  cmdWin->generic.width += widthDiff;
-	  newHeight = screenheight - cmdWin->generic.origin.y;
+	  newHeight = _rl_screenheight - cmdWin->generic.origin.y;
 	  _makeInvisibleAndSetNewHeight (cmdWin, newHeight);
 	  _makeVisibleWithNewHeight (firstWin);
 	  _makeVisibleWithNewHeight (cmdWin);
@@ -716,8 +716,8 @@
 	    newHeight = firstWin->generic.height;
 	  else if ((firstWin->generic.height +
 		    secondWin->generic.height + (splitDiff * 2)) >=
-		   (screenheight - MIN_CMD_WIN_HEIGHT - 1))
-	    newHeight = (screenheight - MIN_CMD_WIN_HEIGHT - 1) / 2;
+		   (_rl_screenheight - MIN_CMD_WIN_HEIGHT - 1))
+	    newHeight = (_rl_screenheight - MIN_CMD_WIN_HEIGHT - 1) / 2;
 	  else if ((firstWin->generic.height + splitDiff) <= 0)
 	    newHeight = MIN_WIN_HEIGHT;
 	  else
@@ -736,9 +736,9 @@
 	    newHeight = secondWin->generic.height;
 	  else if ((firstWin->generic.height +
 		    secondWin->generic.height + (splitDiff * 2)) >=
-		   (screenheight - MIN_CMD_WIN_HEIGHT - 1))
+		   (_rl_screenheight - MIN_CMD_WIN_HEIGHT - 1))
 	    {
-	      newHeight = screenheight - MIN_CMD_WIN_HEIGHT - 1;
+	      newHeight = _rl_screenheight - MIN_CMD_WIN_HEIGHT - 1;
 	      if (newHeight % 2)
 		newHeight = (newHeight / 2) + 1;
 	      else
