--- gdb-6.5/gdb/dwarf2read.c.orig	2006-05-13 17:46:38.000000000 +0200
+++ gdb-6.5/gdb/dwarf2read.c	2006-06-22 15:59:46.254440000 +0200
@@ -6193,9 +6193,11 @@
     case DW_LANG_Modula2:
       cu->language = language_m2;
       break;
+    case DW_LANG_Pascal83:
+      cu->language = language_pascal;
+      break;
     case DW_LANG_Cobol74:
     case DW_LANG_Cobol85:
-    case DW_LANG_Pascal83:
     default:
       cu->language = language_minimal;
       break;
