diff -urN gdb-6.0.org/gdb/gdbserver/Makefile.in gdb-6.0/gdb/gdbserver/Makefile.in
--- gdb-6.0.org/gdb/gdbserver/Makefile.in	2003-10-09 19:42:41.000000000 +0200
+++ gdb-6.0/gdb/gdbserver/Makefile.in	2003-10-09 19:48:08.000000000 +0200
@@ -76,7 +76,7 @@
 
 # Where is the source dir for the READLINE library?  Traditionally in .. or .
 # (For the binary library built from it, we use ${READLINE_DIR}${subdir}.)
-READLINE_DIR = ${srcdir}/../readline
+READLINE_DIR = /usr/lib
 READLINE_DEP = $$(READLINE_DIR)
 
 # All the includes used for CFLAGS and for lint.
diff -urN gdb-6.0.org/gdb/Makefile.in gdb-6.0/gdb/Makefile.in
--- gdb-6.0.org/gdb/Makefile.in	2003-10-09 19:42:39.000000000 +0200
+++ gdb-6.0/gdb/Makefile.in	2003-10-09 19:48:08.000000000 +0200
@@ -116,10 +116,10 @@
 BFD_CFLAGS = -I$(BFD_DIR) -I$(BFD_SRC)
 
 # Where is the READLINE library?  Typically in ../readline.
-READLINE_DIR = ../readline
-READLINE = $(READLINE_DIR)/libreadline.a
-READLINE_SRC = $(srcdir)/$(READLINE_DIR)
-READLINE_CFLAGS = -I$(READLINE_SRC)/..
+READLINE_DIR = /usr/include/readline
+READLINE = -lreadline
+READLINE_SRC = /usr/include/readline
+READLINE_CFLAGS = -I/usr/include/readline
 
 WARN_CFLAGS = @WARN_CFLAGS@
 WERROR_CFLAGS = @WERROR_CFLAGS@
@@ -369,7 +369,7 @@
 	$(XM_CLIBS) $(TM_CLIBS) $(NAT_CLIBS) $(GDBTKLIBS) @LIBS@ \
 	$(LIBICONV) \
 	$(LIBIBERTY) $(WIN32LIBS)
-CDEPS = $(XM_CDEPS) $(TM_CDEPS) $(NAT_CDEPS) $(SIM) $(BFD) $(READLINE) \
+CDEPS = $(XM_CDEPS) $(TM_CDEPS) $(NAT_CDEPS) $(SIM) $(BFD) \
 	$(OPCODES) $(INTL_DEPS) $(LIBIBERTY) $(CONFIG_DEPS)
 
 ADD_FILES = $(XM_ADD_FILES) $(TM_ADD_FILES) $(NAT_ADD_FILES)
--- gdb-6.5/Makefile.in.orig	2006-06-21 07:14:44.000000000 +0200
+++ gdb-6.5/Makefile.in	2006-06-22 15:57:50.144440000 +0200
@@ -598,7 +598,6 @@
     maybe-configure-perl \
     maybe-configure-prms \
     maybe-configure-rcs \
-    maybe-configure-readline \
     maybe-configure-release \
     maybe-configure-recode \
     maybe-configure-sed \
@@ -740,7 +739,6 @@
 all-host: maybe-all-perl
 all-host: maybe-all-prms
 all-host: maybe-all-rcs
-all-host: maybe-all-readline
 all-host: maybe-all-release
 all-host: maybe-all-recode
 all-host: maybe-all-sed
@@ -851,7 +849,6 @@
 info-host: maybe-info-perl
 info-host: maybe-info-prms
 info-host: maybe-info-rcs
-info-host: maybe-info-readline
 info-host: maybe-info-release
 info-host: maybe-info-recode
 info-host: maybe-info-sed
@@ -955,7 +952,6 @@
 dvi-host: maybe-dvi-perl
 dvi-host: maybe-dvi-prms
 dvi-host: maybe-dvi-rcs
-dvi-host: maybe-dvi-readline
 dvi-host: maybe-dvi-release
 dvi-host: maybe-dvi-recode
 dvi-host: maybe-dvi-sed
@@ -1059,7 +1055,6 @@
 html-host: maybe-html-perl
 html-host: maybe-html-prms
 html-host: maybe-html-rcs
-html-host: maybe-html-readline
 html-host: maybe-html-release
 html-host: maybe-html-recode
 html-host: maybe-html-sed
@@ -1163,7 +1158,6 @@
 TAGS-host: maybe-TAGS-perl
 TAGS-host: maybe-TAGS-prms
 TAGS-host: maybe-TAGS-rcs
-TAGS-host: maybe-TAGS-readline
 TAGS-host: maybe-TAGS-release
 TAGS-host: maybe-TAGS-recode
 TAGS-host: maybe-TAGS-sed
@@ -1267,7 +1261,6 @@
 install-info-host: maybe-install-info-perl
 install-info-host: maybe-install-info-prms
 install-info-host: maybe-install-info-rcs
-install-info-host: maybe-install-info-readline
 install-info-host: maybe-install-info-release
 install-info-host: maybe-install-info-recode
 install-info-host: maybe-install-info-sed
@@ -1371,7 +1364,6 @@
 install-html-host: maybe-install-html-perl
 install-html-host: maybe-install-html-prms
 install-html-host: maybe-install-html-rcs
-install-html-host: maybe-install-html-readline
 install-html-host: maybe-install-html-release
 install-html-host: maybe-install-html-recode
 install-html-host: maybe-install-html-sed
@@ -1475,7 +1467,6 @@
 installcheck-host: maybe-installcheck-perl
 installcheck-host: maybe-installcheck-prms
 installcheck-host: maybe-installcheck-rcs
-installcheck-host: maybe-installcheck-readline
 installcheck-host: maybe-installcheck-release
 installcheck-host: maybe-installcheck-recode
 installcheck-host: maybe-installcheck-sed
@@ -1579,7 +1570,6 @@
 mostlyclean-host: maybe-mostlyclean-perl
 mostlyclean-host: maybe-mostlyclean-prms
 mostlyclean-host: maybe-mostlyclean-rcs
-mostlyclean-host: maybe-mostlyclean-readline
 mostlyclean-host: maybe-mostlyclean-release
 mostlyclean-host: maybe-mostlyclean-recode
 mostlyclean-host: maybe-mostlyclean-sed
@@ -1683,7 +1673,6 @@
 clean-host: maybe-clean-perl
 clean-host: maybe-clean-prms
 clean-host: maybe-clean-rcs
-clean-host: maybe-clean-readline
 clean-host: maybe-clean-release
 clean-host: maybe-clean-recode
 clean-host: maybe-clean-sed
@@ -1787,7 +1776,6 @@
 distclean-host: maybe-distclean-perl
 distclean-host: maybe-distclean-prms
 distclean-host: maybe-distclean-rcs
-distclean-host: maybe-distclean-readline
 distclean-host: maybe-distclean-release
 distclean-host: maybe-distclean-recode
 distclean-host: maybe-distclean-sed
@@ -1891,7 +1879,6 @@
 maintainer-clean-host: maybe-maintainer-clean-perl
 maintainer-clean-host: maybe-maintainer-clean-prms
 maintainer-clean-host: maybe-maintainer-clean-rcs
-maintainer-clean-host: maybe-maintainer-clean-readline
 maintainer-clean-host: maybe-maintainer-clean-release
 maintainer-clean-host: maybe-maintainer-clean-recode
 maintainer-clean-host: maybe-maintainer-clean-sed
@@ -2050,7 +2037,6 @@
     maybe-check-perl \
     maybe-check-prms \
     maybe-check-rcs \
-    maybe-check-readline \
     maybe-check-release \
     maybe-check-recode \
     maybe-check-sed \
@@ -2180,7 +2166,6 @@
     maybe-install-perl \
     maybe-install-prms \
     maybe-install-rcs \
-    maybe-install-readline \
     maybe-install-release \
     maybe-install-recode \
     maybe-install-sed \
@@ -2251,7 +2236,6 @@
     maybe-install-perl \
     maybe-install-prms \
     maybe-install-rcs \
-    maybe-install-readline \
     maybe-install-release \
     maybe-install-recode \
     maybe-install-sed \
@@ -40764,7 +40748,6 @@
 all-gdb: maybe-all-libiberty
 all-gdb: maybe-all-opcodes
 all-gdb: maybe-all-bfd
-all-gdb: maybe-all-readline
 all-gdb: maybe-all-build-bison
 all-gdb: maybe-all-build-byacc
 all-gdb: maybe-all-sim
@@ -40993,7 +40976,6 @@
 all-sim: maybe-all-libiberty
 all-sim: maybe-all-bfd
 all-sim: maybe-all-opcodes
-all-sim: maybe-all-readline
 all-sim: maybe-configure-gdb
 all-fastjar: maybe-all-zlib
 all-fastjar: maybe-all-build-texinfo
