diff -ur gdb-5.1.1-/gdb/rs6000-tdep.c gdb-5.1.1/gdb/rs6000-tdep.c
--- gdb-5.1.1-/gdb/rs6000-tdep.c	Sun Jul 15 22:10:02 2001
+++ gdb-5.1.1/gdb/rs6000-tdep.c	Mon Apr  8 18:48:00 2002
@@ -32,9 +32,9 @@
 #include "arch-utils.h"
 #include "regcache.h"
 
-#include "bfd/libbfd.h"		/* for bfd_default_set_arch_mach */
+#include "libbfd.h"		/* for bfd_default_set_arch_mach */
 #include "coff/internal.h"	/* for libcoff.h */
-#include "bfd/libcoff.h"	/* for xcoff_data */
+#include "libcoff.h"		/* for xcoff_data */
 
 #include "elf-bfd.h"
 
