diff -urN gdb-5.0.org/gdb/Makefile.in gdb-5.0/gdb/Makefile.in
--- gdb-5.0.org/gdb/Makefile.in	Tue Sep  5 21:41:32 2000
+++ gdb-5.0/gdb/Makefile.in	Tue Sep  5 21:42:55 2000
@@ -354,6 +354,7 @@
 
 FLAGS_TO_PASS = \
 	"prefix=$(prefix)" \
+	"infodir=$(infodir)" \
 	"exec_prefix=$(exec_prefix)" \
 	"against=$(against)" \
 	"AR=$(AR)" \
--- gdb-6.1.1/gdb/doc/gdb.texinfo.orig	2004-06-15 00:25:29.000000000 +0200
+++ gdb-6.1.1/gdb/doc/gdb.texinfo	2004-06-22 23:37:55.963941600 +0200
@@ -38,9 +38,9 @@
 
 @c This is a dir.info fragment to support semi-automated addition of
 @c manuals to an info tree.
-@dircategory Software development
+@dircategory Programming tools:
 @direntry
-* Gdb: (gdb).                     The GNU debugger.
+* Gdb: (gdb).				The GNU debugger
 @end direntry
 
 @ifinfo
--- gdb-6.1.1/gdb/doc/gdbint.texinfo.orig	2004-06-15 00:25:31.000000000 +0200
+++ gdb-6.1.1/gdb/doc/gdbint.texinfo	2004-06-22 23:38:21.741022888 +0200
@@ -1,9 +1,9 @@
 \input texinfo   @c -*- texinfo -*-
 @setfilename gdbint.info
 @include gdb-cfg.texi
-@dircategory Software development
+@dircategory Programming tools:
 @direntry
-* Gdb-Internals: (gdbint).	The GNU debugger's internals.
+* Gdb-Internals: (gdbint).		The GNU debugger's internals
 @end direntry
 
 @ifinfo
--- gdb-6.1.1/gdb/doc/stabs.texinfo.orig	2004-06-15 00:25:31.000000000 +0200
+++ gdb-6.1.1/gdb/doc/stabs.texinfo	2004-06-22 23:38:51.279532352 +0200
@@ -5,9 +5,9 @@
 
 @c This is a dir.info fragment to support semi-automated addition of
 @c manuals to an info tree.
-@dircategory Software development
+@dircategory Programming tools:
 @direntry
-* Stabs: (stabs).                 The "stabs" debugging information format.   
+* Stabs: (stabs).			The "stabs" debugging information format
 @end direntry
 
 @ifinfo
