--- gdb-6.7.org/gdb/Makefile.in	2007-09-05 02:14:02.000000000 +0200
+++ gdb-6.7/gdb/Makefile.in	2007-10-10 21:11:18.119976680 +0200
@@ -126,11 +126,11 @@
 BFD_CFLAGS = -I$(BFD_DIR) -I$(BFD_SRC)
 
 # Where is the READLINE library?  Typically in ../readline.
-READLINE_DIR = ../readline
-READLINE_SRC = $(srcdir)/$(READLINE_DIR)
-READLINE = @READLINE@
-READLINE_DEPS = @READLINE_DEPS@
-READLINE_CFLAGS = @READLINE_CFLAGS@
+READLINE_DIR = /usr/include/readline
+READLINE = -lreadline
+READLINE_SRC = /usr/include/readline
+READLINE_DEPS =
+READLINE_CFLAGS = -I/usr/include/readline
 
 # Where is expat?  This will be empty if expat was not available.
 LIBEXPAT = @LIBEXPAT@
@@ -586,7 +586,7 @@ CLIBS = $(SIM) $(READLINE) $(OPCODES) $(
 	@LIBS@ @GUILE_LIBS@ @PYTHON_LIBS@ \
 	$(LIBEXPAT) $(LIBLZMA) $(LIBBABELTRACE) $(LIBIPT) \
 	$(LIBIBERTY) $(WIN32LIBS) $(LIBGNU) $(LIBICONV) $(LIBMPFR)
-CDEPS = $(XM_CDEPS) $(NAT_CDEPS) $(SIM) $(BFD) $(READLINE_DEPS) \
+CDEPS = $(XM_CDEPS) $(NAT_CDEPS) $(SIM) $(BFD) \
 	$(OPCODES) $(INTL_DEPS) $(LIBIBERTY) $(CONFIG_DEPS) $(LIBGNU)
 
 ADD_FILES = $(XM_ADD_FILES) $(TM_ADD_FILES) $(NAT_ADD_FILES)
--- gdb-7.4.50.20120120/Makefile.in.orig	2012-01-20 02:48:18.000000000 +0100
+++ gdb-7.4.50.20120120/Makefile.in	2012-03-17 10:35:14.194730128 +0100
@@ -922,7 +922,6 @@
     maybe-configure-libiberty \
     maybe-configure-libiconv \
     maybe-configure-m4 \
-    maybe-configure-readline \
     maybe-configure-sid \
     maybe-configure-sim \
     maybe-configure-texinfo \
@@ -1057,7 +1056,6 @@
 @endif libiberty-no-bootstrap
 all-host: maybe-all-libiconv
 all-host: maybe-all-m4
-all-host: maybe-all-readline
 all-host: maybe-all-sid
 all-host: maybe-all-sim
 all-host: maybe-all-texinfo
@@ -1149,7 +1147,6 @@
 info-host: maybe-info-libiberty
 info-host: maybe-info-libiconv
 info-host: maybe-info-m4
-info-host: maybe-info-readline
 info-host: maybe-info-sid
 info-host: maybe-info-sim
 info-host: maybe-info-texinfo
@@ -1226,7 +1223,6 @@
 dvi-host: maybe-dvi-libiberty
 dvi-host: maybe-dvi-libiconv
 dvi-host: maybe-dvi-m4
-dvi-host: maybe-dvi-readline
 dvi-host: maybe-dvi-sid
 dvi-host: maybe-dvi-sim
 dvi-host: maybe-dvi-texinfo
@@ -1303,7 +1299,6 @@
 pdf-host: maybe-pdf-libiberty
 pdf-host: maybe-pdf-libiconv
 pdf-host: maybe-pdf-m4
-pdf-host: maybe-pdf-readline
 pdf-host: maybe-pdf-sid
 pdf-host: maybe-pdf-sim
 pdf-host: maybe-pdf-texinfo
@@ -1380,7 +1375,6 @@
 html-host: maybe-html-libiberty
 html-host: maybe-html-libiconv
 html-host: maybe-html-m4
-html-host: maybe-html-readline
 html-host: maybe-html-sid
 html-host: maybe-html-sim
 html-host: maybe-html-texinfo
@@ -1457,7 +1451,6 @@
 TAGS-host: maybe-TAGS-libiberty
 TAGS-host: maybe-TAGS-libiconv
 TAGS-host: maybe-TAGS-m4
-TAGS-host: maybe-TAGS-readline
 TAGS-host: maybe-TAGS-sid
 TAGS-host: maybe-TAGS-sim
 TAGS-host: maybe-TAGS-texinfo
@@ -1534,7 +1527,6 @@
 install-info-host: maybe-install-info-libiberty
 install-info-host: maybe-install-info-libiconv
 install-info-host: maybe-install-info-m4
-install-info-host: maybe-install-info-readline
 install-info-host: maybe-install-info-sid
 install-info-host: maybe-install-info-sim
 install-info-host: maybe-install-info-texinfo
@@ -1611,7 +1603,6 @@
 install-pdf-host: maybe-install-pdf-libiberty
 install-pdf-host: maybe-install-pdf-libiconv
 install-pdf-host: maybe-install-pdf-m4
-install-pdf-host: maybe-install-pdf-readline
 install-pdf-host: maybe-install-pdf-sid
 install-pdf-host: maybe-install-pdf-sim
 install-pdf-host: maybe-install-pdf-texinfo
@@ -1688,7 +1679,6 @@
 install-html-host: maybe-install-html-libiberty
 install-html-host: maybe-install-html-libiconv
 install-html-host: maybe-install-html-m4
-install-html-host: maybe-install-html-readline
 install-html-host: maybe-install-html-sid
 install-html-host: maybe-install-html-sim
 install-html-host: maybe-install-html-texinfo
@@ -1765,7 +1755,6 @@
 installcheck-host: maybe-installcheck-libiberty
 installcheck-host: maybe-installcheck-libiconv
 installcheck-host: maybe-installcheck-m4
-installcheck-host: maybe-installcheck-readline
 installcheck-host: maybe-installcheck-sid
 installcheck-host: maybe-installcheck-sim
 installcheck-host: maybe-installcheck-texinfo
@@ -1842,7 +1831,6 @@
 mostlyclean-host: maybe-mostlyclean-libiberty
 mostlyclean-host: maybe-mostlyclean-libiconv
 mostlyclean-host: maybe-mostlyclean-m4
-mostlyclean-host: maybe-mostlyclean-readline
 mostlyclean-host: maybe-mostlyclean-sid
 mostlyclean-host: maybe-mostlyclean-sim
 mostlyclean-host: maybe-mostlyclean-texinfo
@@ -1919,7 +1907,6 @@
 clean-host: maybe-clean-libiberty
 clean-host: maybe-clean-libiconv
 clean-host: maybe-clean-m4
-clean-host: maybe-clean-readline
 clean-host: maybe-clean-sid
 clean-host: maybe-clean-sim
 clean-host: maybe-clean-texinfo
@@ -1996,7 +1983,6 @@
 distclean-host: maybe-distclean-libiberty
 distclean-host: maybe-distclean-libiconv
 distclean-host: maybe-distclean-m4
-distclean-host: maybe-distclean-readline
 distclean-host: maybe-distclean-sid
 distclean-host: maybe-distclean-sim
 distclean-host: maybe-distclean-texinfo
@@ -2073,7 +2059,6 @@
 maintainer-clean-host: maybe-maintainer-clean-libiberty
 maintainer-clean-host: maybe-maintainer-clean-libiconv
 maintainer-clean-host: maybe-maintainer-clean-m4
-maintainer-clean-host: maybe-maintainer-clean-readline
 maintainer-clean-host: maybe-maintainer-clean-sid
 maintainer-clean-host: maybe-maintainer-clean-sim
 maintainer-clean-host: maybe-maintainer-clean-texinfo
@@ -2205,7 +2190,6 @@
     maybe-check-libiberty \
     maybe-check-libiconv \
     maybe-check-m4 \
-    maybe-check-readline \
     maybe-check-sid \
     maybe-check-sim \
     maybe-check-texinfo \
@@ -2308,7 +2292,6 @@
     maybe-install-libiberty \
     maybe-install-libiconv \
     maybe-install-m4 \
-    maybe-install-readline \
     maybe-install-sid \
     maybe-install-sim \
     maybe-install-texinfo \
@@ -2354,7 +2337,6 @@
     maybe-install-libiberty \
     maybe-install-libiconv \
     maybe-install-m4 \
-    maybe-install-readline \
     maybe-install-sid \
     maybe-install-sim \
     maybe-install-texinfo \
@@ -2450,7 +2432,6 @@
     maybe-install-strip-libiberty \
     maybe-install-strip-libiconv \
     maybe-install-strip-m4 \
-    maybe-install-strip-readline \
     maybe-install-strip-sid \
     maybe-install-strip-sim \
     maybe-install-strip-texinfo \
@@ -42608,7 +42589,6 @@
 all-gdb: maybe-all-libiberty
 all-gdb: maybe-all-libiconv
 all-gdb: maybe-all-opcodes
-all-gdb: maybe-all-readline
 all-gdb: maybe-all-build-bison
 all-gdb: maybe-all-sim
 all-gdb: maybe-all-libdecnumber
@@ -42946,7 +42926,6 @@
 all-sim: maybe-all-libiberty
 all-sim: maybe-all-bfd
 all-sim: maybe-all-opcodes
-all-sim: maybe-all-readline
 all-sim: maybe-configure-gdb
 all-fastjar: maybe-all-zlib
 all-fastjar: maybe-all-build-texinfo
