diff -urN gdb-5.0.org/Makefile.in gdb-5.0/Makefile.in
--- gdb-5.0.org/Makefile.in	Wed May 24 18:38:08 2000
+++ gdb-5.0/Makefile.in	Wed May 24 18:40:05 2000
@@ -558,7 +558,6 @@
 	all-perl \
 	all-prms \
 	all-rcs \
-	all-readline \
 	all-release \
 	all-recode \
 	all-sed \
@@ -631,7 +630,6 @@
 	check-perl \
 	check-prms \
 	check-rcs \
-	check-readline \
 	check-recode \
 	check-sed \
 	check-send-pr \
@@ -705,7 +703,6 @@
 	install-perl \
 	install-prms \
 	install-rcs \
-	install-readline \
 	install-recode \
 	install-sed \
 	install-send-pr \
@@ -869,7 +866,6 @@
 	clean-perl \
 	clean-prms \
 	clean-rcs \
-	clean-readline \
 	clean-release \
 	clean-recode \
 	clean-sed \
@@ -1561,7 +1557,7 @@
 all-gcc: all-bison all-byacc all-binutils all-gas all-ld
 all-bootstrap: all-libiberty all-bison all-byacc all-binutils all-gas all-ld
 GDB_TK = all-tk all-tcl all-itcl all-tix all-libgui
-all-gdb: all-libiberty all-opcodes all-bfd all-mmalloc all-readline all-bison all-byacc all-sim $(gdbnlmrequirements) $(GDB_TK)
+all-gdb: all-libiberty all-opcodes all-bfd all-mmalloc all-bison all-byacc all-sim $(gdbnlmrequirements) $(GDB_TK)
 all-gettext:
 all-gnuserv:
 configure-target-gperf: $(ALL_GCC)
@@ -1606,12 +1602,11 @@
 all-perl:
 all-prms: all-libiberty
 all-rcs:
-all-readline:
 all-recode: all-libiberty
 all-sed: all-libiberty
 all-send-pr: all-prms
 all-shellutils:
-all-sim: all-libiberty all-bfd all-opcodes all-readline
+all-sim: all-libiberty all-bfd all-opcodes
 all-tar: all-libiberty
 all-tcl:
 all-tcl8.1:
@@ -1853,7 +1848,7 @@
 		SUPPORT_FILES="$(GNATS_SUPPORT_DIRS)"
 
 .PHONY: gdb.tar.bz2
-GDB_SUPPORT_DIRS= bfd include libiberty mmalloc opcodes readline sim utils intl
+GDB_SUPPORT_DIRS= bfd include libiberty mmalloc opcodes sim utils intl
 gdb.tar.bz2: $(DIST_SUPPORT) $(GDB_SUPPORT_DIRS) gdb
 	$(MAKE) -f Makefile.in gdb-taz TOOL=gdb \
 		MD5PROG="$(MD5PROG)" \
diff -urN gdb-5.0.org/gdb/Makefile.in gdb-5.0/gdb/Makefile.in
--- gdb-5.0.org/gdb/Makefile.in	Wed May 24 18:37:32 2000
+++ gdb-5.0/gdb/Makefile.in	Wed May 24 18:40:47 2000
@@ -126,10 +126,10 @@
 BFD_CFLAGS = -I$(BFD_DIR) -I$(BFD_SRC)
 
 # Where is the READLINE library?  Typically in ../readline.
-READLINE_DIR = ../readline
-READLINE = $(READLINE_DIR)/libreadline.a
-READLINE_SRC = $(srcdir)/$(READLINE_DIR)
-READLINE_CFLAGS = -I$(READLINE_SRC)/..
+READLINE_DIR = /usr/include/readline
+READLINE = -lreadline
+READLINE_SRC = /usr/include/readline
+READLINE_CFLAGS = -I/usr/include/readline
 
 WARN_CFLAGS = @WARN_CFLAGS@
 WERROR_CFLAGS = @WERROR_CFLAGS@
