diff -urN gdb-6.0.org/gdb/gdbserver/Makefile.in gdb-6.0/gdb/gdbserver/Makefile.in
--- gdb-6.0.org/gdb/gdbserver/Makefile.in	2003-10-09 19:42:41.000000000 +0200
+++ gdb-6.0/gdb/gdbserver/Makefile.in	2003-10-09 19:48:08.000000000 +0200
@@ -76,7 +76,7 @@
 
 # Where is the source dir for the READLINE library?  Traditionally in .. or .
 # (For the binary library built from it, we use ${READLINE_DIR}${subdir}.)
-READLINE_DIR = ${srcdir}/../readline
+READLINE_DIR = /usr/lib
 READLINE_DEP = $$(READLINE_DIR)
 
 # All the includes used for CFLAGS and for lint.
diff -urN gdb-6.0.org/gdb/Makefile.in gdb-6.0/gdb/Makefile.in
--- gdb-6.0.org/gdb/Makefile.in	2003-10-09 19:42:39.000000000 +0200
+++ gdb-6.0/gdb/Makefile.in	2003-10-09 19:48:08.000000000 +0200
@@ -116,10 +116,10 @@
 BFD_CFLAGS = -I$(BFD_DIR) -I$(BFD_SRC)
 
 # Where is the READLINE library?  Typically in ../readline.
-READLINE_DIR = ../readline
-READLINE = $(READLINE_DIR)/libreadline.a
-READLINE_SRC = $(srcdir)/$(READLINE_DIR)
-READLINE_CFLAGS = -I$(READLINE_SRC)/..
+READLINE_DIR = /usr/include/readline
+READLINE = -lreadline
+READLINE_SRC = /usr/include/readline
+READLINE_CFLAGS = -I/usr/include/readline
 
 WARN_CFLAGS = @WARN_CFLAGS@
 WERROR_CFLAGS = @WERROR_CFLAGS@
@@ -369,7 +369,7 @@
 	$(XM_CLIBS) $(TM_CLIBS) $(NAT_CLIBS) $(GDBTKLIBS) @LIBS@ \
 	$(LIBICONV) \
 	$(LIBIBERTY) $(WIN32LIBS)
-CDEPS = $(XM_CDEPS) $(TM_CDEPS) $(NAT_CDEPS) $(SIM) $(BFD) $(READLINE) \
+CDEPS = $(XM_CDEPS) $(TM_CDEPS) $(NAT_CDEPS) $(SIM) $(BFD) \
 	$(OPCODES) $(INTL_DEPS) $(LIBIBERTY) $(CONFIG_DEPS)
 
 ADD_FILES = $(XM_ADD_FILES) $(TM_ADD_FILES) $(NAT_ADD_FILES)
--- gdb-6.1/Makefile.in.orig	2004-04-03 18:52:09.000000000 +0200
+++ gdb-6.1/Makefile.in	2004-04-17 22:43:50.603804248 +0200
@@ -571,7 +571,6 @@
     maybe-configure-perl \
     maybe-configure-prms \
     maybe-configure-rcs \
-    maybe-configure-readline \
     maybe-configure-release \
     maybe-configure-recode \
     maybe-configure-sed \
@@ -661,7 +660,6 @@
     maybe-all-perl \
     maybe-all-prms \
     maybe-all-rcs \
-    maybe-all-readline \
     maybe-all-release \
     maybe-all-recode \
     maybe-all-sed \
@@ -755,7 +753,6 @@
     maybe-info-perl \
     maybe-info-prms \
     maybe-info-rcs \
-    maybe-info-readline \
     maybe-info-release \
     maybe-info-recode \
     maybe-info-sed \
@@ -1724,28 +1721,6 @@
 	  || exit 1
 
 
-.PHONY: maybe-info-readline info-readline
-maybe-info-readline:
-
-info-readline: \
-    configure-readline 
-	@[ -f ./readline/Makefile ] || exit 0; \
-	r=`${PWD_COMMAND}`; export r; \
-	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
-	$(SET_LIB_PATH) \
-	for flag in $(EXTRA_HOST_FLAGS); do \
-	  eval `echo "$$flag" | sed -e "s|^\([^=]*\)=\(.*\)|\1='\2'; export \1|"`; \
-	done; \
-	echo "Doing info in readline" ; \
-	(cd readline && \
-	  $(MAKE) $(BASE_FLAGS_TO_PASS) "AR=$${AR}" "AS=$${AS}" \
-	          "CC=$${CC}" "CXX=$${CXX}" "LD=$${LD}" "NM=$${NM}" \
-	          "RANLIB=$${RANLIB}" \
-	          "DLLTOOL=$${DLLTOOL}" "WINDRES=$${WINDRES}" \
-	          info) \
-	  || exit 1
-
-
 .PHONY: maybe-info-release info-release
 maybe-info-release:
 
@@ -2654,7 +2629,6 @@
     maybe-dvi-perl \
     maybe-dvi-prms \
     maybe-dvi-rcs \
-    maybe-dvi-readline \
     maybe-dvi-release \
     maybe-dvi-recode \
     maybe-dvi-sed \
@@ -3623,28 +3597,6 @@
 	  || exit 1
 
 
-.PHONY: maybe-dvi-readline dvi-readline
-maybe-dvi-readline:
-
-dvi-readline: \
-    configure-readline 
-	@[ -f ./readline/Makefile ] || exit 0; \
-	r=`${PWD_COMMAND}`; export r; \
-	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
-	$(SET_LIB_PATH) \
-	for flag in $(EXTRA_HOST_FLAGS); do \
-	  eval `echo "$$flag" | sed -e "s|^\([^=]*\)=\(.*\)|\1='\2'; export \1|"`; \
-	done; \
-	echo "Doing dvi in readline" ; \
-	(cd readline && \
-	  $(MAKE) $(BASE_FLAGS_TO_PASS) "AR=$${AR}" "AS=$${AS}" \
-	          "CC=$${CC}" "CXX=$${CXX}" "LD=$${LD}" "NM=$${NM}" \
-	          "RANLIB=$${RANLIB}" \
-	          "DLLTOOL=$${DLLTOOL}" "WINDRES=$${WINDRES}" \
-	          dvi) \
-	  || exit 1
-
-
 .PHONY: maybe-dvi-release dvi-release
 maybe-dvi-release:
 
@@ -4553,7 +4505,6 @@
     maybe-TAGS-perl \
     maybe-TAGS-prms \
     maybe-TAGS-rcs \
-    maybe-TAGS-readline \
     maybe-TAGS-release \
     maybe-TAGS-recode \
     maybe-TAGS-sed \
@@ -5522,28 +5473,6 @@
 	  || exit 1
 
 
-.PHONY: maybe-TAGS-readline TAGS-readline
-maybe-TAGS-readline:
-
-TAGS-readline: \
-    configure-readline 
-	@[ -f ./readline/Makefile ] || exit 0; \
-	r=`${PWD_COMMAND}`; export r; \
-	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
-	$(SET_LIB_PATH) \
-	for flag in $(EXTRA_HOST_FLAGS); do \
-	  eval `echo "$$flag" | sed -e "s|^\([^=]*\)=\(.*\)|\1='\2'; export \1|"`; \
-	done; \
-	echo "Doing TAGS in readline" ; \
-	(cd readline && \
-	  $(MAKE) $(BASE_FLAGS_TO_PASS) "AR=$${AR}" "AS=$${AS}" \
-	          "CC=$${CC}" "CXX=$${CXX}" "LD=$${LD}" "NM=$${NM}" \
-	          "RANLIB=$${RANLIB}" \
-	          "DLLTOOL=$${DLLTOOL}" "WINDRES=$${WINDRES}" \
-	          TAGS) \
-	  || exit 1
-
-
 .PHONY: maybe-TAGS-release TAGS-release
 maybe-TAGS-release:
 
@@ -6452,7 +6381,6 @@
     maybe-install-info-perl \
     maybe-install-info-prms \
     maybe-install-info-rcs \
-    maybe-install-info-readline \
     maybe-install-info-release \
     maybe-install-info-recode \
     maybe-install-info-sed \
@@ -7463,29 +7391,6 @@
 	  || exit 1
 
 
-.PHONY: maybe-install-info-readline install-info-readline
-maybe-install-info-readline:
-
-install-info-readline: \
-    configure-readline \
-    info-readline 
-	@[ -f ./readline/Makefile ] || exit 0; \
-	r=`${PWD_COMMAND}`; export r; \
-	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
-	$(SET_LIB_PATH) \
-	for flag in $(EXTRA_HOST_FLAGS); do \
-	  eval `echo "$$flag" | sed -e "s|^\([^=]*\)=\(.*\)|\1='\2'; export \1|"`; \
-	done; \
-	echo "Doing install-info in readline" ; \
-	(cd readline && \
-	  $(MAKE) $(BASE_FLAGS_TO_PASS) "AR=$${AR}" "AS=$${AS}" \
-	          "CC=$${CC}" "CXX=$${CXX}" "LD=$${LD}" "NM=$${NM}" \
-	          "RANLIB=$${RANLIB}" \
-	          "DLLTOOL=$${DLLTOOL}" "WINDRES=$${WINDRES}" \
-	          install-info) \
-	  || exit 1
-
-
 .PHONY: maybe-install-info-release install-info-release
 maybe-install-info-release:
 
@@ -8433,7 +8338,6 @@
     maybe-installcheck-perl \
     maybe-installcheck-prms \
     maybe-installcheck-rcs \
-    maybe-installcheck-readline \
     maybe-installcheck-release \
     maybe-installcheck-recode \
     maybe-installcheck-sed \
@@ -9402,28 +9306,6 @@
 	  || exit 1
 
 
-.PHONY: maybe-installcheck-readline installcheck-readline
-maybe-installcheck-readline:
-
-installcheck-readline: \
-    configure-readline 
-	@[ -f ./readline/Makefile ] || exit 0; \
-	r=`${PWD_COMMAND}`; export r; \
-	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
-	$(SET_LIB_PATH) \
-	for flag in $(EXTRA_HOST_FLAGS); do \
-	  eval `echo "$$flag" | sed -e "s|^\([^=]*\)=\(.*\)|\1='\2'; export \1|"`; \
-	done; \
-	echo "Doing installcheck in readline" ; \
-	(cd readline && \
-	  $(MAKE) $(BASE_FLAGS_TO_PASS) "AR=$${AR}" "AS=$${AS}" \
-	          "CC=$${CC}" "CXX=$${CXX}" "LD=$${LD}" "NM=$${NM}" \
-	          "RANLIB=$${RANLIB}" \
-	          "DLLTOOL=$${DLLTOOL}" "WINDRES=$${WINDRES}" \
-	          installcheck) \
-	  || exit 1
-
-
 .PHONY: maybe-installcheck-release installcheck-release
 maybe-installcheck-release:
 
@@ -10332,7 +10214,6 @@
     maybe-mostlyclean-perl \
     maybe-mostlyclean-prms \
     maybe-mostlyclean-rcs \
-    maybe-mostlyclean-readline \
     maybe-mostlyclean-release \
     maybe-mostlyclean-recode \
     maybe-mostlyclean-sed \
@@ -11245,27 +11126,6 @@
 	  || exit 1
 
 
-.PHONY: maybe-mostlyclean-readline mostlyclean-readline
-maybe-mostlyclean-readline:
-
-mostlyclean-readline: 
-	@[ -f ./readline/Makefile ] || exit 0; \
-	r=`${PWD_COMMAND}`; export r; \
-	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
-	$(SET_LIB_PATH) \
-	for flag in $(EXTRA_HOST_FLAGS); do \
-	  eval `echo "$$flag" | sed -e "s|^\([^=]*\)=\(.*\)|\1='\2'; export \1|"`; \
-	done; \
-	echo "Doing mostlyclean in readline" ; \
-	(cd readline && \
-	  $(MAKE) $(BASE_FLAGS_TO_PASS) "AR=$${AR}" "AS=$${AS}" \
-	          "CC=$${CC}" "CXX=$${CXX}" "LD=$${LD}" "NM=$${NM}" \
-	          "RANLIB=$${RANLIB}" \
-	          "DLLTOOL=$${DLLTOOL}" "WINDRES=$${WINDRES}" \
-	          mostlyclean) \
-	  || exit 1
-
-
 .PHONY: maybe-mostlyclean-release mostlyclean-release
 maybe-mostlyclean-release:
 
@@ -12107,7 +11967,6 @@
     maybe-clean-perl \
     maybe-clean-prms \
     maybe-clean-rcs \
-    maybe-clean-readline \
     maybe-clean-release \
     maybe-clean-recode \
     maybe-clean-sed \
@@ -13034,27 +12893,6 @@
 	  || exit 1
 
 
-.PHONY: maybe-clean-readline clean-readline
-maybe-clean-readline:
-
-clean-readline: 
-	@[ -f ./readline/Makefile ] || exit 0; \
-	r=`${PWD_COMMAND}`; export r; \
-	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
-	$(SET_LIB_PATH) \
-	for flag in $(EXTRA_HOST_FLAGS); do \
-	  eval `echo "$$flag" | sed -e "s|^\([^=]*\)=\(.*\)|\1='\2'; export \1|"`; \
-	done; \
-	echo "Doing clean in readline" ; \
-	(cd readline && \
-	  $(MAKE) $(BASE_FLAGS_TO_PASS) "AR=$${AR}" "AS=$${AS}" \
-	          "CC=$${CC}" "CXX=$${CXX}" "LD=$${LD}" "NM=$${NM}" \
-	          "RANLIB=$${RANLIB}" \
-	          "DLLTOOL=$${DLLTOOL}" "WINDRES=$${WINDRES}" \
-	          clean) \
-	  || exit 1
-
-
 .PHONY: maybe-clean-release clean-release
 maybe-clean-release:
 
@@ -13896,7 +13734,6 @@
     maybe-distclean-perl \
     maybe-distclean-prms \
     maybe-distclean-rcs \
-    maybe-distclean-readline \
     maybe-distclean-release \
     maybe-distclean-recode \
     maybe-distclean-sed \
@@ -14823,27 +14660,6 @@
 	  || exit 1
 
 
-.PHONY: maybe-distclean-readline distclean-readline
-maybe-distclean-readline:
-
-distclean-readline: 
-	@[ -f ./readline/Makefile ] || exit 0; \
-	r=`${PWD_COMMAND}`; export r; \
-	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
-	$(SET_LIB_PATH) \
-	for flag in $(EXTRA_HOST_FLAGS); do \
-	  eval `echo "$$flag" | sed -e "s|^\([^=]*\)=\(.*\)|\1='\2'; export \1|"`; \
-	done; \
-	echo "Doing distclean in readline" ; \
-	(cd readline && \
-	  $(MAKE) $(BASE_FLAGS_TO_PASS) "AR=$${AR}" "AS=$${AS}" \
-	          "CC=$${CC}" "CXX=$${CXX}" "LD=$${LD}" "NM=$${NM}" \
-	          "RANLIB=$${RANLIB}" \
-	          "DLLTOOL=$${DLLTOOL}" "WINDRES=$${WINDRES}" \
-	          distclean) \
-	  || exit 1
-
-
 .PHONY: maybe-distclean-release distclean-release
 maybe-distclean-release:
 
@@ -15685,7 +15501,6 @@
     maybe-maintainer-clean-perl \
     maybe-maintainer-clean-prms \
     maybe-maintainer-clean-rcs \
-    maybe-maintainer-clean-readline \
     maybe-maintainer-clean-release \
     maybe-maintainer-clean-recode \
     maybe-maintainer-clean-sed \
@@ -16612,27 +16427,6 @@
 	  || exit 1
 
 
-.PHONY: maybe-maintainer-clean-readline maintainer-clean-readline
-maybe-maintainer-clean-readline:
-
-maintainer-clean-readline: 
-	@[ -f ./readline/Makefile ] || exit 0; \
-	r=`${PWD_COMMAND}`; export r; \
-	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
-	$(SET_LIB_PATH) \
-	for flag in $(EXTRA_HOST_FLAGS); do \
-	  eval `echo "$$flag" | sed -e "s|^\([^=]*\)=\(.*\)|\1='\2'; export \1|"`; \
-	done; \
-	echo "Doing maintainer-clean in readline" ; \
-	(cd readline && \
-	  $(MAKE) $(BASE_FLAGS_TO_PASS) "AR=$${AR}" "AS=$${AS}" \
-	          "CC=$${CC}" "CXX=$${CXX}" "LD=$${LD}" "NM=$${NM}" \
-	          "RANLIB=$${RANLIB}" \
-	          "DLLTOOL=$${DLLTOOL}" "WINDRES=$${WINDRES}" \
-	          maintainer-clean) \
-	  || exit 1
-
-
 .PHONY: maybe-maintainer-clean-release maintainer-clean-release
 maybe-maintainer-clean-release:
 
@@ -17531,7 +17325,6 @@
     maybe-check-perl \
     maybe-check-prms \
     maybe-check-rcs \
-    maybe-check-readline \
     maybe-check-release \
     maybe-check-recode \
     maybe-check-sed \
@@ -17641,7 +17434,6 @@
     maybe-install-perl \
     maybe-install-prms \
     maybe-install-rcs \
-    maybe-install-readline \
     maybe-install-release \
     maybe-install-recode \
     maybe-install-sed \
@@ -17708,7 +17500,6 @@
     maybe-install-perl \
     maybe-install-prms \
     maybe-install-rcs \
-    maybe-install-readline \
     maybe-install-release \
     maybe-install-recode \
     maybe-install-sed \
@@ -20631,73 +20422,6 @@
 	(cd rcs && $(MAKE) $(FLAGS_TO_PASS) install)
 
 
-.PHONY: configure-readline maybe-configure-readline
-maybe-configure-readline:
-configure-readline:
-	@test ! -f readline/Makefile || exit 0; \
-	[ -d readline ] || mkdir readline; \
-	r=`${PWD_COMMAND}`; export r; \
-	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
-	CC="$(CC)"; export CC; \
-	CFLAGS="$(CFLAGS)"; export CFLAGS; \
-	CONFIG_SHELL="$(SHELL)"; export CONFIG_SHELL; \
-	CXX="$(CXX)"; export CXX; \
-	CXXFLAGS="$(CXXFLAGS)"; export CXXFLAGS; \
-	AR="$(AR)"; export AR; \
-	AS="$(AS)"; export AS; \
-	CC_FOR_BUILD="$(CC_FOR_BUILD)"; export CC_FOR_BUILD; \
-	DLLTOOL="$(DLLTOOL)"; export DLLTOOL; \
-	LD="$(LD)"; export LD; \
-	NM="$(NM)"; export NM; \
-	RANLIB="$(RANLIB)"; export RANLIB; \
-	WINDRES="$(WINDRES)"; export WINDRES; \
-	OBJCOPY="$(OBJCOPY)"; export OBJCOPY; \
-	OBJDUMP="$(OBJDUMP)"; export OBJDUMP; \
-	echo Configuring in readline; \
-	cd readline || exit 1; \
-	case $(srcdir) in \
-	  \.) \
-	    srcdiroption="--srcdir=."; \
-	    libsrcdir=".";; \
-	  /* | [A-Za-z]:[\\/]*) \
-	    srcdiroption="--srcdir=$(srcdir)/readline"; \
-	    libsrcdir="$$s/readline";; \
-	  *) \
-	    srcdiroption="--srcdir=../$(srcdir)/readline"; \
-	    libsrcdir="$$s/readline";; \
-	esac; \
-	$(SHELL) $${libsrcdir}/configure \
-	  $(HOST_CONFIGARGS) $${srcdiroption} \
-	  || exit 1
-
-.PHONY: all-readline maybe-all-readline
-maybe-all-readline:
-all-readline: configure-readline
-	@r=`${PWD_COMMAND}`; export r; \
-	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
-	$(SET_LIB_PATH) \
-	(cd readline && $(MAKE) $(FLAGS_TO_PASS) all)
-
-.PHONY: check-readline maybe-check-readline
-maybe-check-readline:
-
-check-readline:
-	@r=`${PWD_COMMAND}`; export r; \
-	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
-	$(SET_LIB_PATH) \
-	(cd readline && $(MAKE) $(FLAGS_TO_PASS) check)
-
-
-.PHONY: install-readline maybe-install-readline
-maybe-install-readline:
-
-install-readline: installdirs
-	@r=`${PWD_COMMAND}`; export r; \
-	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
-	$(SET_LIB_PATH) \
-	(cd readline && $(MAKE) $(FLAGS_TO_PASS) install)
-
-
 .PHONY: configure-release maybe-configure-release
 maybe-configure-release:
 configure-release:
@@ -23771,7 +23495,7 @@
 # GDB needs to know that the simulator is being built.
 configure-gdb: maybe-configure-itcl maybe-configure-tcl maybe-configure-tk maybe-configure-sim
 GDB_TK = @GDB_TK@
-all-gdb: maybe-all-libiberty maybe-all-opcodes maybe-all-bfd maybe-all-mmalloc maybe-all-readline maybe-all-bison maybe-all-byacc maybe-all-sim $(gdbnlmrequirements) $(GDB_TK)
+all-gdb: maybe-all-libiberty maybe-all-opcodes maybe-all-bfd maybe-all-mmalloc maybe-all-bison maybe-all-byacc maybe-all-sim $(gdbnlmrequirements) $(GDB_TK)
 install-gdb: maybe-install-tcl maybe-install-tk maybe-install-itcl maybe-install-tix maybe-install-libgui
 configure-libgui: maybe-configure-tcl maybe-configure-tk
 all-libgui: maybe-all-tcl maybe-all-tk maybe-all-itcl
@@ -23802,7 +23526,7 @@
 install-itcl: maybe-install-tcl
 all-sid: maybe-all-libiberty maybe-all-bfd maybe-all-opcodes maybe-all-tcl maybe-all-tk
 install-sid: maybe-install-tcl maybe-install-tk
-all-sim: maybe-all-libiberty maybe-all-bfd maybe-all-opcodes maybe-all-readline maybe-configure-gdb
+all-sim: maybe-all-libiberty maybe-all-bfd maybe-all-opcodes maybe-configure-gdb
 configure-tk: maybe-configure-tcl
 all-tk: maybe-all-tcl
 configure-tix: maybe-configure-tcl maybe-configure-tk
