--- gdb-6.8.50.20081125/gdb/valprint.c~	2008-11-26 17:29:53.836145817 +0100
+++ gdb-6.8.50.20081125/gdb/valprint.c	2008-11-26 17:31:59.193231994 +0100
@@ -65,12 +65,12 @@
 struct value_print_options user_print_options =
 {
   Val_pretty_default,		/* pretty */
-  0,				/* prettyprint_arrays */
-  0,				/* prettyprint_structs */
-  0,				/* vtblprint */
+  1,				/* prettyprint_arrays */
+  1,				/* prettyprint_structs */
+  1,				/* vtblprint */
   1,				/* unionprint */
   1,				/* addressprint */
-  0,				/* objectprint */
+  1,				/* objectprint */
   PRINT_MAX_DEFAULT,		/* print_max */
   10,				/* repeat_count_threshold */
   0,				/* output_format */
