diff -urN gdb-6.0.org/gdb/config.in gdb-6.0/gdb/config.in
--- gdb-6.0.org/gdb/config.in	2003-10-09 19:42:39.000000000 +0200
+++ gdb-6.0/gdb/config.in	2003-10-09 19:44:01.000000000 +0200
@@ -290,6 +290,9 @@
 /* Define if you have the <ncurses.h> header file.  */
 #undef HAVE_NCURSES_H
 
+/* Define if you have the <ncurses/ncurses.h> header file.  */
+#undef HAVE_NCURSES_NCURSES_H
+
 /* Define if you have the <ndir.h> header file.  */
 #undef HAVE_NDIR_H
 
diff -urN gdb-6.0.org/gdb/configure.in gdb-6.0/gdb/configure.in
--- gdb-6.0.org/gdb/configure.in	2003-10-09 19:42:39.000000000 +0200
+++ gdb-6.0/gdb/configure.in	2003-10-09 19:44:27.000000000 +0200
@@ -351,7 +351,7 @@
    Solaris 2.[78] when using GCC. ]])
     fi ;;
 esac
-AC_CHECK_HEADERS(curses.h ncurses.h term.h)
+AC_CHECK_HEADERS(curses.h ncurses.h ncurses/ncurses.h term.h)
 
 # FIXME: kettenis/20030102: In most cases we include these
 # unconditionally, so what's the point in checking these?
diff -urN gdb-6.0.org/gdb/tui/tuiData.h gdb-6.0/gdb/tui/tuiData.h
--- gdb-6.0.org/gdb/tui/tuiData.h	2003-10-09 19:42:43.000000000 +0200
+++ gdb-6.0/gdb/tui/tuiData.h	2003-10-09 19:44:01.000000000 +0200
@@ -24,6 +24,8 @@
 
 #if defined (HAVE_NCURSES_H)
 #include <ncurses.h>
+#elif defined (HAVE_NCURSES_NCURSES_H)
+#include <ncurses/ncurses.h>
 #elif defined (HAVE_CURSES_H)
 #include <curses.h>
 #endif
diff -urN gdb-6.0.org/gdb/utils.c gdb-6.0/gdb/utils.c
--- gdb-6.0.org/gdb/utils.c	2003-10-09 19:42:40.000000000 +0200
+++ gdb-6.0/gdb/utils.c	2003-10-09 19:44:01.000000000 +0200
@@ -52,9 +52,14 @@
 
 #include <sys/param.h>		/* For MAXPATHLEN */
 
-#ifdef HAVE_CURSES_H
+#if defined (HAVE_NCURSES_H)
+#include <ncurses.h>
+#elif defined (HAVE_NCURSES_NCURSES_H)
+#include <ncurses/ncurses.h>
+#elif defined (HAVE_CURSES_H)
 #include <curses.h>
 #endif
+
 #ifdef HAVE_TERM_H
 #include <term.h>
 #endif
