diff -ur gdb-6.7.org/gdb/Makefile.in gdb-6.7/gdb/Makefile.in
--- gdb-6.7.org/gdb/Makefile.in	2007-09-05 02:14:02.000000000 +0200
+++ gdb-6.7/gdb/Makefile.in	2007-10-10 21:11:18.119976680 +0200
@@ -126,11 +126,11 @@
 BFD_CFLAGS = -I$(BFD_DIR) -I$(BFD_SRC)
 
 # Where is the READLINE library?  Typically in ../readline.
-READLINE_DIR = ../readline
-READLINE_SRC = $(srcdir)/$(READLINE_DIR)
-READLINE = @READLINE@
-READLINE_DEPS = @READLINE_DEPS@
-READLINE_CFLAGS = @READLINE_CFLAGS@
+READLINE_DIR = /usr/include/readline
+READLINE = -lreadline
+READLINE_SRC = /usr/include/readline
+READLINE_DEPS =
+READLINE_CFLAGS = -I/usr/include/readline
 
 # Where is expat?  This will be empty if expat was not available.
 LIBEXPAT = @LIBEXPAT@
@@ -416,7 +416,7 @@
 	$(XM_CLIBS) $(NAT_CLIBS) $(GDBTKLIBS) @LIBS@ \
 	$(LIBICONV) $(LIBEXPAT) \
 	$(LIBIBERTY) $(WIN32LIBS) $(LIBGNU)
-CDEPS = $(XM_CDEPS) $(NAT_CDEPS) $(SIM) $(BFD) $(READLINE_DEPS) \
+CDEPS = $(XM_CDEPS) $(NAT_CDEPS) $(SIM) $(BFD) \
 	$(OPCODES) $(INTL_DEPS) $(LIBIBERTY) $(CONFIG_DEPS) $(LIBGNU)
 
 ADD_FILES = $(XM_ADD_FILES) $(TM_ADD_FILES) $(NAT_ADD_FILES)
diff -ur gdb-6.7.org/Makefile.in gdb-6.7/Makefile.in
--- gdb-6.7.org/Makefile.in	2007-10-10 19:09:32.000000000 +0200
+++ gdb-6.7/Makefile.in	2007-10-10 21:10:23.310556251 +0200
@@ -664,7 +664,6 @@
     maybe-configure-perl \
     maybe-configure-prms \
     maybe-configure-rcs \
-    maybe-configure-readline \
     maybe-configure-release \
     maybe-configure-recode \
     maybe-configure-sed \
@@ -812,7 +811,6 @@
 all-host: maybe-all-perl
 all-host: maybe-all-prms
 all-host: maybe-all-rcs
-all-host: maybe-all-readline
 all-host: maybe-all-release
 all-host: maybe-all-recode
 all-host: maybe-all-sed
@@ -927,7 +925,6 @@
 info-host: maybe-info-perl
 info-host: maybe-info-prms
 info-host: maybe-info-rcs
-info-host: maybe-info-readline
 info-host: maybe-info-release
 info-host: maybe-info-recode
 info-host: maybe-info-sed
@@ -1033,7 +1030,6 @@
 dvi-host: maybe-dvi-perl
 dvi-host: maybe-dvi-prms
 dvi-host: maybe-dvi-rcs
-dvi-host: maybe-dvi-readline
 dvi-host: maybe-dvi-release
 dvi-host: maybe-dvi-recode
 dvi-host: maybe-dvi-sed
@@ -1245,7 +1241,6 @@
 html-host: maybe-html-perl
 html-host: maybe-html-prms
 html-host: maybe-html-rcs
-html-host: maybe-html-readline
 html-host: maybe-html-release
 html-host: maybe-html-recode
 html-host: maybe-html-sed
@@ -1351,7 +1346,6 @@
 TAGS-host: maybe-TAGS-perl
 TAGS-host: maybe-TAGS-prms
 TAGS-host: maybe-TAGS-rcs
-TAGS-host: maybe-TAGS-readline
 TAGS-host: maybe-TAGS-release
 TAGS-host: maybe-TAGS-recode
 TAGS-host: maybe-TAGS-sed
@@ -1457,7 +1451,6 @@
 install-info-host: maybe-install-info-perl
 install-info-host: maybe-install-info-prms
 install-info-host: maybe-install-info-rcs
-install-info-host: maybe-install-info-readline
 install-info-host: maybe-install-info-release
 install-info-host: maybe-install-info-recode
 install-info-host: maybe-install-info-sed
@@ -1669,7 +1662,6 @@
 install-html-host: maybe-install-html-perl
 install-html-host: maybe-install-html-prms
 install-html-host: maybe-install-html-rcs
-install-html-host: maybe-install-html-readline
 install-html-host: maybe-install-html-release
 install-html-host: maybe-install-html-recode
 install-html-host: maybe-install-html-sed
@@ -1775,7 +1767,6 @@
 installcheck-host: maybe-installcheck-perl
 installcheck-host: maybe-installcheck-prms
 installcheck-host: maybe-installcheck-rcs
-installcheck-host: maybe-installcheck-readline
 installcheck-host: maybe-installcheck-release
 installcheck-host: maybe-installcheck-recode
 installcheck-host: maybe-installcheck-sed
@@ -1881,7 +1872,6 @@
 mostlyclean-host: maybe-mostlyclean-perl
 mostlyclean-host: maybe-mostlyclean-prms
 mostlyclean-host: maybe-mostlyclean-rcs
-mostlyclean-host: maybe-mostlyclean-readline
 mostlyclean-host: maybe-mostlyclean-release
 mostlyclean-host: maybe-mostlyclean-recode
 mostlyclean-host: maybe-mostlyclean-sed
@@ -1987,7 +1977,6 @@
 clean-host: maybe-clean-perl
 clean-host: maybe-clean-prms
 clean-host: maybe-clean-rcs
-clean-host: maybe-clean-readline
 clean-host: maybe-clean-release
 clean-host: maybe-clean-recode
 clean-host: maybe-clean-sed
@@ -2093,7 +2082,6 @@
 distclean-host: maybe-distclean-perl
 distclean-host: maybe-distclean-prms
 distclean-host: maybe-distclean-rcs
-distclean-host: maybe-distclean-readline
 distclean-host: maybe-distclean-release
 distclean-host: maybe-distclean-recode
 distclean-host: maybe-distclean-sed
@@ -2199,7 +2187,6 @@
 maintainer-clean-host: maybe-maintainer-clean-perl
 maintainer-clean-host: maybe-maintainer-clean-prms
 maintainer-clean-host: maybe-maintainer-clean-rcs
-maintainer-clean-host: maybe-maintainer-clean-readline
 maintainer-clean-host: maybe-maintainer-clean-release
 maintainer-clean-host: maybe-maintainer-clean-recode
 maintainer-clean-host: maybe-maintainer-clean-sed
@@ -2359,7 +2346,6 @@
     maybe-check-perl \
     maybe-check-prms \
     maybe-check-rcs \
-    maybe-check-readline \
     maybe-check-release \
     maybe-check-recode \
     maybe-check-sed \
@@ -2491,7 +2477,6 @@
     maybe-install-perl \
     maybe-install-prms \
     maybe-install-rcs \
-    maybe-install-readline \
     maybe-install-release \
     maybe-install-recode \
     maybe-install-sed \
@@ -2564,7 +2549,6 @@
     maybe-install-perl \
     maybe-install-prms \
     maybe-install-rcs \
-    maybe-install-readline \
     maybe-install-release \
     maybe-install-recode \
     maybe-install-sed \
@@ -51494,7 +51478,6 @@
 all-gdb: maybe-all-intl
 all-gdb: maybe-all-libiberty
 all-gdb: maybe-all-opcodes
-all-gdb: maybe-all-readline
 all-gdb: maybe-all-build-bison
 all-gdb: maybe-all-build-byacc
 all-gdb: maybe-all-sim
@@ -51826,7 +51809,6 @@
 all-sim: maybe-all-libiberty
 all-sim: maybe-all-bfd
 all-sim: maybe-all-opcodes
-all-sim: maybe-all-readline
 all-sim: maybe-configure-gdb
 all-fastjar: maybe-all-zlib
 all-fastjar: maybe-all-build-texinfo
--- gdb-6.8.50.20090224/gdb/tui/tui-io.c~	2009-02-24 09:17:18.617891690 +0100
+++ gdb-6.8.50.20090224/gdb/tui/tui-io.c	2009-02-24 09:17:43.554606911 +0100
@@ -512,7 +512,7 @@
 void
 tui_setup_io (int mode)
 {
-  extern int readline_echoing_p;
+  extern int _rl_echoing_p;
  
   if (mode)
     {
@@ -522,12 +522,12 @@
       tui_old_rl_prep_terminal = rl_prep_term_function;
       tui_old_rl_getc_function = rl_getc_function;
       tui_old_rl_outstream = rl_outstream;
-      tui_old_readline_echoing_p = readline_echoing_p;
+      tui_old_readline_echoing_p = _rl_echoing_p;
       rl_redisplay_function = tui_redisplay_readline;
       rl_deprep_term_function = tui_deprep_terminal;
       rl_prep_term_function = tui_prep_terminal;
       rl_getc_function = tui_getc;
-      readline_echoing_p = 0;
+      _rl_echoing_p = 0;
       rl_outstream = tui_rl_outstream;
       rl_prompt = 0;
       rl_completion_display_matches_hook = tui_rl_display_match_list;
@@ -564,7 +564,7 @@
       rl_getc_function = tui_old_rl_getc_function;
       rl_outstream = tui_old_rl_outstream;
       rl_completion_display_matches_hook = 0;
-      readline_echoing_p = tui_old_readline_echoing_p;
+      _rl_echoing_p = tui_old_readline_echoing_p;
       rl_already_prompted = 0;
 
       /* Save tty for SIGCONT.  */
