--- gdb-11.1/gdb/nat/amd64-linux-siginfo.c.orig	2021-07-03 19:41:10.000000000 +0200
+++ gdb-11.1/gdb/nat/amd64-linux-siginfo.c	2021-11-21 18:32:29.634785279 +0100
@@ -330,6 +330,7 @@
       to->cpt_si_pid = from_ptrace.cpt_si_pid;
       to->cpt_si_uid = from_ptrace.cpt_si_uid;
     }
+#ifndef __ILP32__
   else if (to->si_code == SEGV_BNDERR
 	   && to->si_signo == SIGSEGV)
     {
@@ -337,6 +338,7 @@
       to->cpt_si_lower = from_ptrace.cpt_si_lower;
       to->cpt_si_upper = from_ptrace.cpt_si_upper;
     }
+#endif
   else if (to->si_code < 0)
     {
       to->cpt_si_pid = from_ptrace.cpt_si_pid;
