--- gdb-8.0/gdb/gdbserver/linux-amd64-ipa.c~	2017-06-04 17:51:26.000000000 +0200
+++ gdb-8.0/gdb/gdbserver/linux-amd64-ipa.c	2017-06-07 01:14:20.673962263 +0200
@@ -182,8 +182,8 @@
       return tdesc_x32_linux;
     case X86_TDESC_AVX:
       return tdesc_x32_avx_linux;
-    case X86_TDESC_AVX512:
-      return tdesc_x32_avx512_linux;
+    case X86_TDESC_AVX_AVX512:
+      return tdesc_x32_avx_avx512_linux;
     default:
       break;
     }
@@ -279,7 +279,7 @@
 #if defined __ILP32__
   init_registers_x32_linux ();
   init_registers_x32_avx_linux ();
-  init_registers_x32_avx512_linux ();
+  init_registers_x32_avx_avx512_linux ();
 #else
   init_registers_amd64_linux ();
   init_registers_amd64_avx_linux ();
