--- gdb-8.0/gdb/gdbserver/linux-amd64-ipa.c~	2017-06-04 17:51:26.000000000 +0200
+++ gdb-8.0/gdb/gdbserver/linux-amd64-ipa.c	2017-06-07 01:14:20.673962263 +0200
@@ -182,8 +182,8 @@
       return tdesc_x32_linux;
     case X86_TDESC_AVX:
       return tdesc_x32_avx_linux;
-    case X86_TDESC_AVX512:
-      return tdesc_x32_avx512_linux;
+    case X86_TDESC_AVX_AVX512:
+      return tdesc_x32_avx_avx512_linux;
     default:
       break;
     }
@@ -279,7 +279,7 @@
 #if defined __ILP32__
   init_registers_x32_linux ();
   init_registers_x32_avx_linux ();
-  init_registers_x32_avx512_linux ();
+  init_registers_x32_avx_avx512_linux ();
 #else
   init_registers_amd64_linux ();
   init_registers_amd64_avx_linux ();
--- /home/users/baggins/devel/PLD/rpm/BUILD/gdb-8.0/gdb/gdbserver/configure.srv~	2017-06-04 17:51:26.000000000 +0200
+++ /home/users/baggins/devel/PLD/rpm/BUILD/gdb-8.0/gdb/gdbserver/configure.srv	2017-06-08 07:56:18.541252390 +0200
@@ -31,6 +31,7 @@
 
 ipa_i386_linux_regobj="i386-linux-ipa.o i386-avx-linux-ipa.o  i386-avx-mpx-linux-ipa.o i386-avx-avx512-linux-ipa.o i386-avx-mpx-avx512-pku-linux-ipa.o i386-mpx-linux-ipa.o i386-mmx-linux-ipa.o"
 ipa_amd64_linux_regobj="amd64-linux-ipa.o amd64-avx-linux-ipa.o amd64-avx-mpx-linux-ipa.o amd64-avx-avx512-linux-ipa.o amd64-avx-mpx-avx512-pku-linux-ipa.o amd64-mpx-linux-ipa.o"
+ipa_x32_linux_regobj="x32-linux-ipa.o x32-avx-linux-ipa.o x32-avx-avx512-linux-ipa.o"
 ipa_ppc_linux_regobj="powerpc-32l-ipa.o powerpc-altivec32l-ipa.o powerpc-cell32l-ipa.o powerpc-vsx32l-ipa.o powerpc-isa205-32l-ipa.o powerpc-isa205-altivec32l-ipa.o powerpc-isa205-vsx32l-ipa.o powerpc-e500l-ipa.o powerpc-64l-ipa.o powerpc-altivec64l-ipa.o powerpc-cell64l-ipa.o powerpc-vsx64l-ipa.o powerpc-isa205-64l-ipa.o powerpc-isa205-altivec64l-ipa.o powerpc-isa205-vsx64l-ipa.o"
 
 srv_i386_32bit_xmlfiles="i386/32bit-core.xml i386/32bit-sse.xml i386/32bit-avx.xml i386/32bit-avx512.xml i386/32bit-mpx.xml i386/32bit-pkeys.xml"
